pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarStreamMetadataProvider.java

```diff
@@ -80,16 +84,15 @@ public StreamPartitionMsgOffset fetchStreamPartitionOffset(OffsetCriteria offset
     Preconditions.checkNotNull(offsetCriteria);
     try {
       MessageId offset = null;
+      Consumer consumer =
```

Comment:
we're reading earliest/largest in so many places now..
In `PulsarConfig`:
Based on `OffsetCriteria` smallest/largest/custom, we get 
`InitialMessageID` earliest/latest/custom
Based on `InitialMessageID`, we get 
`SubscriberInitialPosition` Earliest/Latest/Latest.
Is it okay to have `SubscriberInitialPosition`=Latest for a custom OffsetCriteria? Should we even keep custom offset criteria in Pulsar? I see in `fetchStreamPartitionOffset` we don't support it.

Further, in `PulsarStreamMetadataProvider#fetchStreamPartitionOffset`, the Consumer is created based on `SubscriberInitialPosition`. But based on the `offsetCriteria` param in that method, we make the consumer `getLastMessage` or `receive` message. What if `PulsarConfig` had `largest` `OffsetCriteria`, but the param in this method has `smallest`?

Reply:
Removed the case for custom offset handling and honouring the provided OffsetCriteria argument. This should address your comment.